koans/8-AboutApplyingWhatWeHaveLearnt.test.js

```diff
@@ -75,15 +77,24 @@ describe("About Applying What We Have Learnt", function() {
         }
     }
 
-    expect(ingredientCount['mushrooms']).toBe(FILL_ME_IN);
+    expect(ingredientCount['mushrooms']).toBe(2);
   });
 
   it("should count the ingredient occurrence (functional)", function () {
     var ingredientCount = { "{ingredient name}": 0 };
 
     /* chain() together map(), flatten() and reduce() */
-
-    expect(ingredientCount['mushrooms']).toBe(FILL_ME_IN);
+    ingredientCount = _(products).chain()
+        .map( function(x) { return x.ingredients })
+        .flatten()
+        .reduce( function (counter, ingredient) {
+            if (counter[ingredient] == null) counter[ingredient] = 0
```

Comment:
actually, not really:
![image](https://user-images.githubusercontent.com/22149132/28007234-450ec54c-655b-11e7-8ebb-57d13e6fad9b.png)